src/libasr/pass/unique_symbols.cpp

```diff
@@ -46,16 +46,29 @@ class SymbolRenameVisitor: public ASR::BaseWalkVisitor<SymbolRenameVisitor> {
     bool all_symbols_mangling;
     bool bindc_mangling = false;
     bool fortran_mangling;
+    bool c_mangling;
     bool should_mangle = false;
     std::vector<std::string> parent_function_name;
     std::string module_name = "";
     SymbolTable* current_scope = nullptr;
 
-    SymbolRenameVisitor(bool mm, bool gm, bool im, bool am, bool bcm, bool fm) :
+    SymbolRenameVisitor(bool mm, bool gm, bool im, bool am, bool bcm, bool fm, bool cm) :
     module_name_mangling(mm), global_symbols_mangling(gm), intrinsic_symbols_mangling(im),
-    all_symbols_mangling(am), bindc_mangling(bcm), fortran_mangling(fm) {}
+    all_symbols_mangling(am), bindc_mangling(bcm), fortran_mangling(fm) , c_mangling(cm){}
 
+    
+    const std::unordered_set<std::string> restricted_keywords_c = {
+        "_Alignas", "_Alignof", "_Atomic", "_Bool", "_Complex", "_Generic", "_Imaginary", "_Noreturn", "_Static_assert", "_Thread_local", "auto", "break", "case", "char", "_Bool", "const", "continue", "default", "do", "double", "else", "enum", "extern", "float", "for", "goto", "if", "int", "long", "register", "return", "short", "signed", "sizeof", "static", "struct", "switch", "typedef", "union", "unsigned", "void", "volatile", "while"
+    };
```

Comment:
These needs to be passed in as an argument later, by the ASR->C backend. It should not live in this file.

Reply:
The `unique_symbols` pass is currently not called by `asr_to_c`. It is called in `lpython.cpp` , by a method declared in `pass_manager`. It would be difficult to take the list of reserved keywords from `asr_to_c`  as all the passes are called before the `asr_to_c` in `lpython.cpp`.
Here is a snippet from the function `emit_c` :
```
    pass_manager.use_default_passes(true);
    compiler_options.po.always_run = true;
    compiler_options.po.run_fun = "f";
    compiler_options.po.c_mangling = true;

    pass_manager.apply_passes(al, asr, compiler_options.po, diagnostics);

    diagnostics.diagnostics.clear();
    auto res = LCompilers::asr_to_c(al, *asr, diagnostics, compiler_options, 0);
```
I am not sure how to restructure this to accommodate the passing of restricted keywords from `asr_to_c` or any other equivalent step to the `unique_symbols` pass.

Reply:
Ok. Until we refactor the pass infrastructure, we can just hardwire C mangling in there, like you did.

Reply:
We'll have to figure out a way to only do this for the C backend, but not normally. One way or the other.

Reply:
Add a TODO there.